src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -581,5 +560,42 @@ private static int SerializeSanitizedCategoryName(byte[] buffer, int cursor, str
 
         return cursor;
     }
+
+    private class ExporterStateForScopes
+    {
+        internal ushort ScopeDepth;
+        internal int IndexForArrayLength;
+        internal int Cursor;
+        internal byte[] Buffer;
```

Comment:
If you made this...

```csharp
    internal byte[] Buffer = new byte[BUFFER_SIZE];
```

You could probably get rid of...

```csharp
private static readonly ThreadLocal<byte[]> m_buffer = new ThreadLocal<byte[]>(() => null);
```

...and just use `exporterState` for everything the exporter needs on a thread.